x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBlobCacheService.java

```diff
@@ -1021,18 +775,16 @@ public boolean tryRead(ByteBuffer buf, long offset) throws IOException {
             var fileRegion = lastAccessedRegion;
             if (fileRegion != null && fileRegion.chunk.regionKey.region == startRegion) {
                 // existing item, check if we need to promote item
-                long now = threadPool.relativeTimeInMillis();
-                if (now - fileRegion.lastAccessed >= minTimeDelta) {
-                    maybePromote(now, fileRegion);
-                }
+                fileRegion.touch();
+
             } else {
-                fileRegion = get(cacheKey, length, startRegion);
+                fileRegion = cache.get(cacheKey, length, startRegion);
             }
-            final var region = fileRegion.chunk;
-            if (region.tracker.checkAvailable(end - getRegionStart(startRegion)) == false) {
+            final var region = fileRegion;
```

Comment:
Why change this? :) Seems we're increasing the number of field accesses here, not sure it matters much but seems unnecessary?

Reply:
Think that was a leftover from a previous version, I've changed it back again